### PR TITLE
Add WB, WBWI Create, UpdateTimestamp, Iterator::Refresh in C API

### PR DESCRIPTION
**Pull request by muthukrishnan24**
This PR adds UpdateTimestamp API of WriteBatch and WBWI, create WB, WBWI with all options and Iterator Refresh in C API

### CODE REVIEW

**Comment by muthukrishnan24** [comment]
@riversand963

**Comment by muthukrishnan24** [comment]
@ajkr, @riversand963 can you please review this PR?

**Comment by muthukrishnan24** [comment]
@riversand963 / @ajkr , can you please review this PR?

File: HISTORY.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # Rocksdb Change Log
 ## Unreleased
 
+### Public API changes
+* Add `rocksdb_writebatch_update_timestamps`, `rocksdb_writebatch_wi_update_timestamps` in C API.
+* Add `rocksdb_iter_refresh` in C API.
+* Add `rocksdb_writebatch_create_with_params`, `rocksdb_writebatch_wi_create_with_params` to create WB and WBWI with all options in C API
```
Comment (ajkr) [inline HISTORY.md:7]: We have a new way of adding release notes; see "unreleased_history/README.txt" after updating against main branch.

Although, it isn't strictly required to have release notes for new C APIs, so simply deleting the notes here would also work. I'd do it myself, but don't have access to edit this branch.

**Review by ajkr — CHANGES_REQUESTED** [review]
Thanks! Branch needs to be updated. Looks good otherwise.

File: HISTORY.md (hunk 1 above)
Comment (muthukrishnan24) [reply HISTORY.md]: Changed, please check

File: include/rocksdb/c.h [hunk 2]
```diff
@@ -850,6 +859,16 @@ rocksdb_writebatch_wi_create(size_t reserved_bytes,
                              unsigned char overwrite_keys);
 extern ROCKSDB_LIBRARY_API rocksdb_writebatch_wi_t*
 rocksdb_writebatch_wi_create_from(const char* rep, size_t size);
+extern ROCKSDB_LIBRARY_API rocksdb_writebatch_wi_t*
+rocksdb_writebatch_wi_create(size_t reserved_bytes,
+                             unsigned char overwrite_keys);
+extern ROCKSDB_LIBRARY_API rocksdb_writebatch_wi_t*
+rocksdb_writebatch_wi_create_from(const char* rep, size_t size);
```
Comment (ajkr) [inline include/rocksdb/c.h:866]: These lines are duplicates of above

**Review by ajkr — CHANGES_REQUESTED** [review]
One more small thing. Thanks for reviving this.

**Review by ajkr — APPROVED** [review]
LGTM!

**Pull request closed without merging**